app/models/customer.rb

```diff
@@ -0,0 +1,13 @@
+class Customer < ApplicationRecord
+  validates :name, presence: true
+  validates :registered_at, presence: true
+  validates :address, presence: true
+  validates :city, presence: true
+  validates :state, presence: true
+  validates :postal_code, presence: true
```

Comment:
You could consider validating whether the data you're getting for these fields is consistent with what's expected. For example, is the postal code 6 digits? Big picture, you want to always validate that what is required exists, and if it exists, that the format (data type, length, etc)  is what you want it to be.